Windows sbt-launch-lib.bash doesn't allow home directory with spaces

- [x] I've read the [CONTRIBUTING](https://github.com/sbt/sbt/blob/develop/CONTRIBUTING.md) guidelines

## steps

On Windows:

* Create an account with a space in it (e.g. `Eric Peters`)
* Install Windows Subsystem Linux (I've installed debian)
* Install java11 (this might also work for java9) - this problem doesn't appear with java8 code path in 
* Set your home directory to your windows home directory (e.g. `/c/Users/Eric Peters`)
* Install sbt in WSL (`apt-get install sbt`)
* Run `sbt` in WSL

## problem
```
$ sbt -v
[process_args] java_version = '11'
[copyRt] java9_rt = '/c/Users/Eric Peters/.sbt/0.13/java9-rt-ext-oracle_corporation_11_0_1/rt.jar'
# Executing command line:
java
-Xms1024m
-Xmx1024m
-XX:ReservedCodeCacheSize=128m
-XX:MaxMetaspaceSize=256m
-XX:+UseParallelGC
-Dscala.ext.dirs=/c/Users/Eric
Peters/.sbt/0.13/java9-rt-ext-oracle_corporation_11_0_1
-jar
/usr/share/sbt/bin/sbt-launch.jar

Error: Could not find or load main class Peters..sbt.0.13.java9-rt-ext-oracle_corporation_11_0_1"
Caused by: java.lang.ClassNotFoundException: Peters//sbt/0/13/java9-rt-ext-oracle_corporation_11_0_1"
```

## expectation


```
$ java \
    -Xms1024m \
    -Xmx1024m \
    -XX:ReservedCodeCacheSize=128m \
    -XX:MaxMetaspaceSize=256m \
    -XX:+UseParallelGC \
    -Dscala.ext.dirs="/c/Users/Eric Peters/.sbt/0.13/java9-rt-ext-oracle_corporation_11_0_1" \
    -jar /usr/share/sbt/bin/sbt-launch.jar
[info] Loading settings for project global-plugins from idea.sbt ...
[info] Loading global plugins from /c/Users/Eric Peters/.sbt/1.0/plugins
[info] Loading project definition from /c/Users/Eric Peters/project
[info] Set current project to eric-peters (in build file:/c/Users/Eric%20Peters/)
[info] sbt server started at local:///c/Users/Eric Peters/.sbt/1.0/server/3bf131f37fd1632b10ff/sock
sbt:eric-peters>
```

This doesn't happen if set to JDK8.

## notes

sbt version: `1.2.8`
java version: `11.0.1`

--- a/src/universal/bin/sbt-launch-lib.bash
+++ b/src/universal/bin/sbt-launch-lib.bash
@@ -57,14 +57,10 @@ rt_export_file () {
 }
 
 execRunner () {
-  # print the arguments one to a line, quoting any containing spaces
+  # print the arguments one to a line, shell-quoted
   [[ $verbose || $debug ]] && echo "# Executing command line:" && {
-    for arg; do
-      if printf "%s\n" "$arg" | grep -q ' '; then
-        printf "\"%s\"\n" "$arg"
-      else
-        printf "%s\n" "$arg"
-      fi
+    for arg in "$@"; do
+       printf "%q\n" "$arg"
     done
     echo ""
   }
@@ -348,7 +344,7 @@ run() {
     $(get_gc_opts) \
     ${JAVA_OPTS} \
     ${SBT_OPTS:-$default_sbt_opts} \
-    ${java_args[@]} \
+    "${java_args[@]}" \
     -jar "$sbt_jar" \
     "${sbt_commands[@]}" \
     "${residual_args[@]}"

